source/dofs/dof_handler_policy.cc

```diff
@@ -1764,7 +1757,10 @@ namespace internal
                 // previously assigned a number to (i.e. the ones on
                 // the interface)
                 local_dof_indices.resize(cell->get_fe().n_dofs_per_cell());
-                cell->get_dof_indices(local_dof_indices);
+                internal::DoFAccessorImplementation::Implementation::
+                  get_dof_indices(*cell,
+                                  local_dof_indices,
+                                  cell->active_fe_index());
```

Comment:
I think this change is necessary because we are not updating the cache as many times, but I don't understand it - the comment above this implies that the cell has dofs, but now we don't call `cell->get_dof_indices()`. Could you add some explanations here and elsewhere?

Reply:
yes, it is because we have not yet populated the cache - I will add a comment here and elsewhere (once I switch back to the branch).

Reply:
I now added comments about the fact that the cache is not ready, which should become obvious now also as one reads through the code in that particular file with all implementations.